From: Matthew Garrett <matthew.garrett@nebula.com>
Subject: [PATCH V3 11/11] Add option to automatically enforce module signatures when in Secure Boot mode
Date: Tue,  3 Sep 2013 19:50:18 -0400
Patch-mainline: Never, SUSE-Xen specific
References: fate#314486
Target: SLE-12

UEFI Secure Boot provides a mechanism for ensuring that the firmware will
only load signed bootloaders and kernels. Certain use cases may also
require that all kernel modules also be signed. Add a configuration option
that enforces this automatically when enabled.

Signed-off-by: Matthew Garrett <matthew.garrett@nebula.com>
Acked-by: Lee, Chun-Yi <jlee@suse.com>

Automatically created from "patches.suse/0011_V3_Add_option_to_automatically_enforce_module_signatures_when_in_Secure_Boot_mode.patch" by xen-port-patches.py

--- a/arch/x86/kernel/setup-xen.c
+++ b/arch/x86/kernel/setup-xen.c
@@ -1276,6 +1276,13 @@ void __init setup_arch(char **cmdline_p)
 
 	io_delay_init();
 
+#ifdef CONFIG_EFI_SECURE_BOOT_SIG_ENFORCE
+	if (efi_enabled(EFI_SECURE_BOOT)) {
+		enforce_signed_modules();
+		pr_info("Secure boot enabled\n");
+	}
+#endif
+
 #ifdef CONFIG_ACPI
 	if (!is_initial_xendomain()) {
 		printk(KERN_INFO "ACPI in unprivileged domain disabled\n");
